src/Platform/Microsoft.Testing.Platform.MSBuild/Tasks/NamespaceHelpers.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Testing.Platform.MSBuild;
+
+internal static class NamespaceHelpers
+{
+    internal static string ToSafeNamespace(string value)
```

Comment:
Please add a comment specifying this is copied/inspired from template code.

Reply:
Maybe also the commit or date it's taken from so that if they do some bug fix we can easily compare dates/commits.

Reply:
```suggestion
    // https://github.com/dotnet/templating/blob/b0b1283f8c96be35f1b65d4b0c1ec0534d86fc2f/src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/ValueForms/DefaultSafeNamespaceValueFormFactory.cs#L17-L59
    internal static string ToSafeNamespace(string value)
```